File: internal/symbol/testdata/src/sample.go
package sample

import (
	"fmt"
	"strings"
)

// Widget is a UI element.
// It carries a name.
type Widget struct {
	Name string
}

// Render draws the widget and returns its label.
func (w Widget) Render() string {
	return decorate(w.Name)
}

// decorate wraps s in brackets.
func decorate(s string) string {
	return fmt.Sprintf("[%s]", strings.ToUpper(s))
}

// Build makes a Widget.
func Build() *Widget { return &Widget{} }

// MaxWidgets caps the pool.
const MaxWidgets = 10

var defaultWidget = Widget{}

File: internal/symbol/testdata/src/sample.ts
/** A widget class. */
export class Widget {
  name: string;

  /** Draws the widget. */
  render(): string {
    return brackets(this.name);
  }
}

// brackets wraps s in square brackets.
function brackets(s: string): string {
  return `[${s}]`;
}

export const MAX = 10;
